polkadot/runtime/common/src/paras_registrar/mod.rs

```diff
@@ -431,6 +507,76 @@ pub mod pallet {
 			runtime_parachains::set_current_head::<T>(para, new_head);
 			Ok(())
 		}
+
+		/// Changes the billing account of a parachain to the caller.
+		///
+		/// ## Arguments
+		/// - `origin`: Can be called by Root, the parachain, or the parachain manager if the
+		///   parachain is unlocked.
+		/// - `para`: The parachain's ID for which the billing account is set.
+		///
+		/// ## Deposits/Fees
+		/// For this call to be successful, the caller account must have a sufficient balance
+		/// to cover the current deposit required for this parachain.
+		///
+		/// ## Events
+		/// The `BillingAccountSet` event is emitted in case of success.
+		#[pallet::call_index(9)]
+		#[pallet::weight(<T as Config>::WeightInfo::set_parachain_billing_account_to_self())]
+		pub fn set_parachain_billing_account_to_self(
```

Comment:
Maybe being able to "whitelist" some kind account by the para for being fees would also be nice, but not really a requirement for now.